text/0000-private-registry-auth.md

```diff
@@ -0,0 +1,130 @@
+- Feature Name: N/A
+- Start Date: 2019-07-05
+- RFC PR: [rust-lang/rfcs#2719](https://github.com/rust-lang/rfcs/pull/2719)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+The ability to authenticate for all requests on private registries is added.
+This includes API requests, crate downloads, and index retrievals.
+The global authentication is enabled in Cargo's configuration in a registry definition.
+
+# Motivation
+[motivation]: #motivation
+
+Businesses with projects in Rust will need somewhere to keep their crates, and the current
+implementation of custom registries only sends authentication at times when it is needed on
+crates.io. There is no option to send the token when downloading crates or making other
+API calls.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+If a custom registry is to be considered private, the registry can require
+authentication on all requests. This can be enabled on the server side by
+modifying the registry definition in the index:
+```json
+{
+    "dl": "https://my-intranet.local/api/v1/crates",
+    "api": "https://my-intranet.local/",
+    "auth_required": true,
+    "token_url": "https://my-intranet.local/me"
```

Comment:
Cargo already assumes a [`/me`](https://doc.rust-lang.org/cargo/reference/registries.html#login) URL for this purpose, what's the benefit of allowing it to be redefined here?

Reply:
This allows for customization in case the API for this registry server is different.

Reply:
To add, this defaults to `/me` if not specified.